docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,297 @@
+   +----+----+               ^
+   | VolMgr  |---bind------> :
+   +----+----+               v
+        |              +------------------------+
+        +---watch----> | PersistentVolumeClaim  | <--+
+                       +------------------------+    |
+                                 ^                   |
+                                 |         +---------+-- user-created
+                                 |         |
+                           +----------+    |
+                           |    Pod   | <--+
+                           +----------+
+```
+
+### PersistentVolumeClaim
+
+A `PersistentVolumeClaim` does not specify which specific PersistentVolume it
+wants or even which volume plugin should be used.  Instead it specifies
+characteristics of the storage it needs and those characteristics are satisfied
+as best is possible.  Characteristics include the minimum storage space
+required and the minimum access mode required (e.g. multi-writer).
+
+As the ecosystem of volume plugins grows, there will be an ever-increasing
+number of things that could be specified about volumes.  Rather than growing
+`PersistentVolumeClaim` to cover the union of every volume's parameters, users
+will be able to request coarse "classes" of volume (e.g. Gold, Silver, Bronze).
+How those classes map to real storage technologies is a policy set by the
+cluster administrators.
+
+### VolumeManagers
+
+The act of binding a `PersistentVolumeClaim` to a `PersistentVolume` is the
+responsibility of a process in the cluster.  A cluster may have any number of
+`VolumeManager`s running, each of which handles some set of claim classes.
+When a `PersistentVolumeClaim` is created that asks for "Gold" storage, the
+`VolumeManager` which understands the "Gold" class will make the binding.
+
+`VolumeManager`s can be simple `Pod`s running in the cluster, pulling
+cofiguration from hardcoded values, flags, config objects, or whatever
+mechanism makes sense for them.  We expect config objects to be a perfect fit
+for this use case.
+
+This very loose coupling means that an admin can define classes however they
+like.  For example, "Gold" might mean GCE PersistentDisk of type SSD, "Silver"
+might mean GCE PersistentDisk of type Disk, and "Bronze" might mean NFS.
+Configuring the same `VolumeManager` to run for both "Gold" and "Silver" with
+just a change in params is trivial, as would be adding a totally bespoke
+`VolumeManager`.
+
+Note that classes are just identifiers - there are no special names or
+meanings.  One of the `VolumeManager`s must be the "default" for claims that do
+not specify a class.
+
+### Config
+
+We expect that `VolumeManager`s will need some amount of configuration, and it
+will be convenient if they have similar parameters, but this should be by
+convention, not by diktat.  Example parameters:
+   * Minimum free instances - how many ready-to-use volumes should be maintained
+   * Maximum free instances - limit of number of ready-to-use volumes
```

Comment:
Would it make more sense for this to be "maximum instances" rather than "maximum free instances"? It seems people might be more likely to want to limit the total size of the pool (allocated + unallocated) rather than just limiting unallocated.